src/simplify_algebra.cpp

```diff
@@ -72,14 +74,39 @@ struct find_mul_conv
         auto a_ins    = r.instructions["a"];
         auto w_ins    = r.instructions["w"];
 
-        auto broadcast_op = any_cast<op::broadcast>(a_ins->get_operator());
-        if(broadcast_op.axis != 1)
+        auto a_input_lens = a_ins->inputs().front()->get_shape().lens();
+
+        std::size_t num_not_one_dims = std::count_if(
+            a_input_lens.cbegin(), a_input_lens.cend(), [](auto dim) { return dim != 1; });
+        if(num_not_one_dims > 1)
             return;
 
+        // check broadcasted along channels
+        auto a_lens    = a_ins->get_shape().lens();
+        auto a_strides = a_ins->get_shape().strides();
+        // handle case of stride = 1 and len = 1
+        auto invalid_sl = [&](std::size_t i) {
+            if(a_strides[i] != 0)
+            {
+                if(a_strides[i] == 1 and a_lens[i] == 1)
+                    return false;
+                return true;
+            }
+            return false;
+        };
```

Comment:
```suggestion
        auto invalid_sl = [&](std::size_t i) {
            if(a_strides[i] > 1)
            {  
                return true;
            } else if(a_strides[i] == 0) { 
              if(a_lens[i] !=1) 
                  return true;
            } else {
            return false;  
            } 
            return false;
        };
```
Should it be something like this ?

Reply:
I am not sure if it is possible but if there is case like `a_lens = [256, 1, 1]`, `a_strides = [256, 1 or 0, 1 or 0]`

Reply:
> That case would be skipped in the current code because a_strides[0] is not 0 or 1. The tensor would also have pretty weird data, as it would not be contiguous.

Sorry, I meant to say this case : `a_lens = [256, 1, 1], a_strides = [1, 1 or 0, 1 or 0]`

Reply:
> . I guess it is possible for a length 1 dimension to have any stride and still be valid? 

Yes i think, the broadcast would work on that.